lib/mp4/probe.js

```diff
@@ -370,6 +371,30 @@ getTracks = function(init) {
   return tracks;
 };
 
+/**
+ * Returns an array of emsg ID3 data from the provided segmentData
+ * ptOffset can also be provided if necessary.
+ */
+getEmsgID3 = function(segmentData, ptOffset = 0) {
+  var emsgBoxes = findBox(segmentData, ['emsg']);
+  var emsgID3Data = [];
+  emsgBoxes.forEach(function(data) {
+      var boxData = new Uint8Array(data);
+      var parsedBox = emsg.parseEmsgBox(boxData);
+      var pts = emsg.scaleTime(parsedBox.presentation_time, 
+                          parsedBox.timescale, 
+                          parsedBox.presentation_time_delta, 
+                          ptOffset);
+      var duration = emsg.scaleTime(parsedBox.event_duration, parsedBox.timescale);
+      emsgID3Data.push({
+          pts,
+          duration,
+          data: parsedBox.message_data,
+      });
+  });
+  return emsgID3Data;
```

Comment:
We can simplify with a map:
`return emsgBoxes.map((data) => { ... return { pts, duration, data }})`

Reply:
Good call on using `map`, I often operate in the mode of trying to write code that "fits" into the codebase, rather than using the most suitable syntax for the task.